app/src/main/java/com/twilio/video/app/ui/login/FirebaseAuthActivity.kt

```diff
@@ -0,0 +1,68 @@
+package com.twilio.video.app.ui.login
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Button
+import androidx.appcompat.app.AppCompatActivity
+import com.firebase.ui.auth.AuthUI
+import com.firebase.ui.auth.AuthUI.IdpConfig.EmailBuilder
+import com.firebase.ui.auth.AuthUI.IdpConfig.GoogleBuilder
+import com.firebase.ui.auth.IdpResponse
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.firebase.auth.FirebaseAuth
+import com.twilio.video.app.R
+import timber.log.Timber
+
+const val RC_SIGN_IN = 20
+
+class FirebaseAuthActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_firebase_auth)
+        findViewById<Button>(R.id.firebase_button).setOnClickListener {
+            navigateToFirebaseUIAuth()
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        if (requestCode == RC_SIGN_IN) {
+            val response = IdpResponse.fromResultIntent(data)
+
+            if (resultCode == Activity.RESULT_OK) {
+                // Successfully signed in
+                val user = FirebaseAuth.getInstance().currentUser
+                Timber.d("Logged in was successful!!")
+                // ...
+            } else {
+                Timber.d("Logged in was not successful :(")
+                // Sign in failed. If response is null the user canceled the
+                // sign-in flow using the back button. Otherwise check
+                // response.getError().getErrorCode() and handle the error.
+                // ...
```

Comment:
Is this a TODO? Do you plan to check the error code to do something more specific?

Reply:
Ah yeah sorry for the confusion. This class no longer exists. I was going back and forth on where to put the FirebaseUI builder and ended up deciding to just include it in the LoginActivity class.